stock_inventory_discrepancy/models/stock_inventory.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class StockInventory(models.Model):
+    _inherit = 'stock.inventory'
+
+    INVENTORY_STATE_SELECTION = [
+        ('draft', 'Draft'),
+        ('cancel', 'Cancelled'),
+        ('confirm', 'In Progress'),
+        ('pending', 'Pending to Approve'),
+        ('done', 'Validated')]
+
+    state = fields.Selection(
+        selection=INVENTORY_STATE_SELECTION,
+        string='Status', readonly=True, index=True, copy=False,
+        help="States of the Inventory Adjustment:\n"
+             "- Draft: Inventory not started.\n"
+             "- In Progress: Inventory in execution.\n"
+             "- Pending to Approve: Inventory have some discrepancies "
+             "greater than the predefined threshold and it's waiting for the "
+             "Control Manager approval.\n"
+             "- Validated: Inventory Approved.")
+    over_discrepancy_line_count = fields.Integer(
+        string='Number of Discrepancies Over Threshold',
+        compute='_compute_over_discrepancy_line_count',
+        store=True)
+
+    @api.multi
+    @api.depends('line_ids.product_qty', 'line_ids.theoretical_qty')
+    def _compute_over_discrepancy_line_count(self):
+        for inventory in self:
+            lines = inventory.line_ids
+            inventory.over_discrepancy_line_count = sum(
+                d.discrepancy_percent > d.discrepancy_threshold
+                for d in lines)
```

Comment:
I'm not too fond of this use of the automatic conversion of booleans to integers, since it is not obvious at first read how this works (for a `_count` method, I expect to see a `len()` not a `sum()`)

Can this be rewritten for instance as:

```
def _compute_over_discrepancy_line_count(self):
        for inventory in self:
            lines = inventory.line_ids.filtered(
                lambda line: line.discrepancy_percent > line.discrepancy_threshold
            )
            inventory.over_discrepancy_line_count = len(lines)
```